spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java

```diff
@@ -170,8 +174,26 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 		else {
 			String[] provisionedDestinations = Stream
 					.of(StringUtils.tokenizeToStringArray(name, ",", true, true))
-					.map(destination -> doProvisionConsumerDestination(destination, group,
-							properties).getName())
+					.flatMap(destination -> {
+						if (properties.isPartitioned() && !ObjectUtils.isEmpty(properties.getInstanceIndexList())) {
+							List<String> consumerDestinationNames = new ArrayList<>();
+
+							for (Integer index : properties.getInstanceIndexList()) {
+								ExtendedConsumerProperties<RabbitConsumerProperties> temporaryProperties =
+										new ExtendedConsumerProperties<>(properties.getExtension());
+								BeanUtils.copyProperties(properties, temporaryProperties);
```

Comment:
We can do this just once, outside the loop.

Reply:
Wanted to be sure there will be no references between objects in-case of something parallel implemented in the future. If that thought is invalid then I can move it outside yes.

Reply:
No that's fine; this is only during provisioning so it's not really important.

Merging.